### PR TITLE
esp32: Expose gpio_set_drive_capability API to Lua

### PR DESCRIPTION
**Pull request by tomsci**
Addresses #3098.

- [x] This PR is for the ~`dev`~ `dev-esp32` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well.
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/*`.

As discussed on the issue, this PR exposes the API to set drive strength on a GPIO pad so its maximum current output can be increased (or decreased).

### CODE REVIEW

**Comment by jpeletier** [comment]
Hi @TOMSCI. Nice one.

My only comment would be to put this as part of `gpio.setup()`, rather than having a separate function.

It could be a table parameter such as `strength` or `drive_strength`. If not specified, then it would not be configured and left as-is, so the call is backward compatible.

**Comment by tomsci** [comment]
@jpeletier I wondered about adding it to `gpio.config()`, but decided against it, because `gpio.config()` maps pretty directly to the underlying Espressif SDK function `gpio_config()` so adding things to it which weren't in the underlying C API didn't feel right. In addition it's possible that you'd want to set the drive strength for a pin which you're not actually using as a GPIO, for example a SPI pin or one configured for use by the PWM or RMT hardware, so making it a parameter to `gpio.config()` would not work for that case (because you don't call `gpio.config()` in those situations). Of course having it in the `gpio` at all is a bit weird when you can use it with non-GPIO-configured pins, but that's what the C API does as well, so I guess there isn't a better place for it.

Basically, it's a separate API in the underlying C, and I think it's probably better to keep it separate in the Lua API too.
